src/esrs/aggregate.rs

```diff
@@ -110,80 +115,40 @@ pub trait AggregateManager: Identifier {
         }
     }
 
-    /// Persists an event into the event store - recording it in the aggregate instance's history.
-    async fn persist(
+    /// Transactional persists events in store - recording it in the aggregate instance's history.
+    /// The store will also project the events. If an error occurs whilst persisting the events,
+    /// the whole transaction is rolled back and the error is returned.
+    ///
+    /// The policies associated to the store are run here. A failure at this point will be silently
+    /// ignored, and the new state returned successfully anyway.
+    ///
+    /// You should _avoid_ implementing this function, and be _very_ careful if you decide to do so.
+    /// The only scenario where this function needs to be overwritten is if you need to change the
+    /// behaviour of policies, e.g. if you want to log something on error.
+    async fn store_events(
         &self,
-        aggregate_state: AggregateState<Self::State>,
+        aggregate_state: &AggregateState<Self::State>,
         events: Vec<Self::Event>,
-    ) -> Result<AggregateState<Self::State>, Self::Error> {
-        let events = self
+    ) -> Result<Vec<StoreEvent<Self::Event>>, Self::Error> {
+        let events: Vec<StoreEvent<Self::Event>> = self
             .event_store()
             .persist(aggregate_state.id, events, aggregate_state.next_sequence_number())
             .await?;
 
         self.event_store().run_policies(&events).await?;
```

Comment:
We should probably think of how to handle policies. If a policy fails here, we return an error instead of the events (that we have recorded nonetheless).

Reply:
Yes, totally agree